### PR TITLE
Add approx1_*_v2 functions to the unified backend

### PR DESCRIPTION
**Pull request by umar456**

### CODE REVIEW

File: src/api/unified/symbol_manager.cpp [hunk 1]
```diff
@@ -246,6 +246,14 @@ bool checkArray(af_backend activeBackend, af_array a) {
     return backend == activeBackend;
 }
 
+bool checkArray(af_backend activeBackend, const af_array* a) {
```
Comment (mark-poscablo) [inline src/api/unified/symbol_manager.cpp]: I like this, it makes checking the output array (whether it exists yet or not) cleaner.

File: src/api/unified/signal.cpp [hunk 2]
```diff
@@ -13,22 +13,36 @@
...
+                     const af_interp_type method, const float offGrid) {
+    CHECK_ARRAYS(yo, yi, xo);
     return CALL(yo, yi, xo, method, offGrid);
 }
 
 af_err af_approx2(af_array *zo, const af_array zi, const af_array xo,
                   const af_array yo, const af_interp_type method,
                   const float offGrid) {
-    CHECK_ARRAYS(zi, xo, yo);
+    CHECK_ARRAYS(zo, zi, xo, yo);
```
Comment (mark-poscablo) [inline src/api/unified/signal.cpp]: My changes for adding the _v2 versions of `af_approx2`/`af_approx2_uniform` are still underway (no PR yet). Should we include this change in this PR now? Alternately, I can add them in the changes that I'm working on right now.

Comment (umar456) [reply src/api/unified/signal.cpp]: This is working on the original function. you don't have to update that in your PR. You are going to be adding the v2 functions.

**Comment by WilliamTambellini** [comment]
tks guys

**Pull request merged**